packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -222,4 +280,26 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
         return Number(value);
     }
 
+    inspect<T>(preferenceName: string, resourceUri?: string): {
+        preferenceName: string,
+        values: Map<PreferenceScope, T>
```

Comment:
each object is JS is already a map, following API like would be nicer and aligned with VS Code:
```ts
{
     preferenceName: string,
     defaultValue: T,
     globalValue: T, // USER
     workspaceValue: T,
     workspaceFolderValue: T,
}
```

Reply:
i updated this function to match the vscode API